applications/NXxas.nxdl.xml

```diff
@@ -22,112 +22,207 @@
+            <item value="keV"/>
+            <item value="degrees"/>
+            <item value="angstrom"/>
+          </enumeration>
+        </field>
+
+        <group type="NXcrystal">
+          <doc>
+            Crystal used for monochromator:
+            chemical_formula, d_spacing, and reflection should be specified
+          </doc>
+          <field name="chemical_formula"></field>
+          <field name="d_spacing" type="NX_FLOAT" units="NX_LENGTH">
+            <doc>spacing between crystal planes of the reflection, in Angstroms</doc>
+          </field>
+          <field name="reflection" type="NX_INT" units="NX_UNITLESS">
+            <doc>Miller indices (hkl) values of nominal reflection</doc>
+            <dimensions><dim index="1" value="3"/></dimensions>
           </field>
         </group>
+      </group>
+      <group type="NXdetector" name="i0">
+        <doc>monitoring the intensity of the incident beam.</doc>
+        <field name="data" type="NX_NUMBER">
+          <dimensions rank="1">
+            <dim index="1" value="nP" />
+          </dimensions>
+        </field>
+      </group>
+
+      <group type="NXdetector" name="itrans">
+        <doc>monitoring the intensity of the transmitted beam.</doc>
+        <field name="data" type="NX_NUMBER">
+          <dimensions rank="1">
+            <dim index="1" value="nP" />
+          </dimensions>
+        </field>
+      </group>
+
+      <group type="NXdetector" name="ifluor">
+        <doc>monitoring the intensity of the fluorescence or emitted beam.</doc>
+        <field name="data" type="NX_NUMBER">
+          <dimensions rank="1">
+            <dim index="1" value="nP" />
+          </dimensions>
+        </field>
+        <field name="mode">
+          <doc>Type of fluorescence or emission signal detected (pick one from enumerated list)</doc>
+          <enumeration>
+            <item value="Total Fluorescence Yield" />
+            <item value="Partial Fluorescence Yield" />
+            <item value="Total Electron Yield" />
+            <item value="Partial Electron Yield" />
+            <item value="Other" />
+          </enumeration>
+        </field>
+      </group>
+
+      <group type="NXdetector" name="irefer">
+        <doc>monitoring the intensity of the reference channel beam</doc>
+        <field name="data" type="NX_NUMBER">
+          <dimensions rank="1">
+            <dim index="1" value="nP" />
+          </dimensions>
+        </field>
+        <field name="mode">
+          <doc>Type of reference channel used (pick one from enumerated list)</doc>
+          <enumeration>
+            <item value="Transmission, downstream of Itrans" />
+            <item value="Fluoresscence, upstream of sample" />
+          </enumeration>
+        </field>
+
+      </group>
+    </group>
+
+    <group type="NXsample">
+      <field name="name">
+        <doc>Descriptive name of sample</doc>
+      </field>
+    </group>
+
+    <group type="NXscan">
```

Comment:
`NXscan` is an application definition, not a base class. If I'm not mistaken, you cannot use an application definition inside another application definition. Maybe `NXxas` could extend `NXscan` @benajamin ?

Reply:
From what I see, there is no need for NXscan here. Just put those field directly under NXxas.

Reply:
The intention here was to add infomation and data about the scan performed to collect the data, just as `<group type="NXsample">` immediately above this would be used to describe the sample. 

Yes, the scan here derives from and extends the base class.   Like, that "NXscan" simply does not have enough data to describe a scan.